VimeoUpload/Descriptor System/KeyedArchiver.swift

```diff
@@ -65,3 +68,23 @@ public class KeyedArchiver: ArchiverProtocol
         return URL.absoluteString! as String
     }
 }
+
+// MARK: NSKeyedArchiver Migration
+
+private extension KeyedArchiver
```

Comment:
I love the idea of intercepting on this level instead on in `load` inside the model class it leaves the model object blind as it should be. 

I also worry that we could potentially not catch edge cases if we only intercept here, whereas on the model level, any time we encounter an instance of the legacy model object (whether it's in cache or download store) we will catch the problem (say there's a crazy case where someone has `VIMVideoFile` cached on their device before updating to v6.1.4)